File: android/app/src/main/java/app/releaf/mobile/features/home/HomeQuickCaptureSection.kt
/*
 * HomeQuickCaptureSection.kt
 * Six quick-capture pills under the library card — one per capture
 * mode the notepad supports: notes, photos, scans, voice, todos,
 * contacts. Tap routes to the notepad tab (future iteration will
 * route through to a pre-selected capture mode on the new entry).
 */

package app.releaf.mobile.features.home

import androidx.compose.foundation.background
import androidx.compose.foundation.border
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.CameraAlt
import androidx.compose.material.icons.filled.EditNote
import androidx.compose.material.icons.filled.CheckBox
import androidx.compose.material.icons.filled.DocumentScanner
import androidx.compose.material.icons.filled.LocationOn
import androidx.compose.material.icons.filled.Mic
import androidx.compose.material3.HorizontalDivider
import androidx.compose.material3.Icon
import androidx.compose.material3.Text
import androidx.compose.material3.VerticalDivider
import androidx.compose.foundation.layout.IntrinsicSize
import androidx.compose.foundation.layout.height
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.vector.ImageVector
import androidx.compose.ui.text.font.FontFamily
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import app.releaf.mobile.ui.theme.AppAccent
import app.releaf.mobile.ui.theme.AppColors
import app.releaf.mobile.ui.theme.AppRadius
import app.releaf.mobile.ui.theme.AppSpacing
import app.releaf.mobile.ui.theme.AppTypography
import app.releaf.mobile.ui.theme.LocalFontWeight

enum class QuickCaptureMode { Notes, Photos, Scans, Voice, Todos, Location }

private data class PillSpec(
    val mode: QuickCaptureMode,
    val label: String,
    val icon: ImageVector,
)

@Composable
fun HomeQuickCaptureSection(
    onCapture: (QuickCaptureMode) -> Unit,
    counts: Map<QuickCaptureMode, Int> = emptyMap(),
    modifier: Modifier = Modifier,
) {
    val specs = listOf(
        PillSpec(QuickCaptureMode.Notes,    "Notes",    Icons.Filled.EditNote),
        PillSpec(QuickCaptureMode.Photos,   "Photos",   Icons.Filled.CameraAlt),
        PillSpec(QuickCaptureMode.Scans,    "Scans",    Icons.Filled.DocumentScanner),
        PillSpec(QuickCaptureMode.Voice,    "Voice",    Icons.Filled.Mic),
        PillSpec(QuickCaptureMode.Todos,    "Todos",    Icons.Filled.CheckBox),
        PillSpec(QuickCaptureMode.Location, "Location", Icons.Filled.LocationOn),
    )
    val rows = specs.chunked(3)

    Column(
        modifier = modifier
            .fillMaxWidth()
            .clip(RoundedCornerShape(AppRadius.lg))
            .background(AppColors.CardSolid)
            .border(1.dp, AppColors.BorderDefault, RoundedCornerShape(AppRadius.lg))
            .padding(AppSpacing.s5),
        verticalArrangement = Arrangement.spacedBy(AppSpacing.s4),
    ) {
        Text(
            text = "HIGHLIGHT",
            style = AppTypography.Eyebrow,
            color = AppColors.ThemeGreenDeep,
        )
        Column {
            rows.forEachIndexed { rowIndex, row ->
                if (rowIndex > 0) {
                    HorizontalDivider(
                        thickness = 1.dp,
                        color     = AppColors.BorderDefault,
                    )
                }
                Row(
                    modifier = Modifier
                        .fillMaxWidth()
                        .height(IntrinsicSize.Min),
                ) {
                    row.forEachIndexed { colIndex, spec ->
                        if (colIndex > 0) {
                            VerticalDivider(
                                thickness = 1.dp,
                                color     = AppColors.BorderDefault,
                            )
                        }
                        Pill(
                            spec    = spec,
                            count   = counts[spec.mode] ?: 0,
                            onClick = { onCapture(spec.mode) },
                            modifier = Modifier.weight(1f),
                        )
                    }
                }
            }
        }
    }
}

@Composable
private fun Pill(
    spec: PillSpec,
    count: Int,
    onClick: () -> Unit,
    modifier: Modifier = Modifier,
) {
    Column(
        modifier = modifier
            .clickable { onClick() }
            .padding(vertical = AppSpacing.s3),
        horizontalAlignment = Alignment.CenterHorizontally,
        verticalArrangement = Arrangement.spacedBy(AppSpacing.s2),
    ) {
        Icon(
            imageVector = spec.icon,
            contentDescription = spec.label,
            tint = AppAccent.primary,
            modifier = Modifier.size(32.dp),
        )
        Text(
            text = spec.label.uppercase(),
            style = AppTypography.Eyebrow,
            color = AppColors.TextSecondary,
            maxLines = 1,
        )
        Text(
            text = "$count",
            color = if (count > 0) AppColors.TextPrimary else AppColors.TextTertiary,
            fontSize = 26.sp,
            fontWeight = LocalFontWeight.current,
            fontFamily = FontFamily.Serif,
            maxLines = 1,
        )
    }
}

File: apps/releaf/ios/Releaf/Features/Home/HomeQuickCaptureSection.swift
/*
 * HomeQuickCaptureSection.swift
 * Six quick-capture pills under the library card — one per capture
 * mode the notepad supports: notes, photos, scans, voice, todos,
 * contacts. Tap routes to the notepad tab (future iteration will
 * route through to a pre-selected capture mode on the new entry).
 *
 * Pure presentation today — onCapture receives the tapped mode and
 * the screen decides what to do with it.
 */

import SwiftUI
import ReleafDesignSystem

public enum QuickCaptureMode: String, CaseIterable, Identifiable {
    case notes, photos, scans, voice, todos, location
    public var id: String { rawValue }

    var label: String {
        switch self {
        case .notes:    return "Notes"
        case .photos:   return "Photos"
        case .scans:    return "Scans"
        case .voice:    return "Voice"
        case .todos:    return "Todos"
        case .location: return "Location"
        }
    }

    var symbol: String {
        switch self {
        case .notes:    return "square.and.pencil"
        case .photos:   return "photo"
        case .scans:    return "doc.viewfinder"
        case .voice:    return "mic"
        case .todos:    return "checklist"
        case .location: return "mappin"
        }
    }
}

public struct HomeQuickCaptureSection: View {
    let counts: [QuickCaptureMode: Int]
    let onCapture: (QuickCaptureMode) -> Void

    public init(
        counts: [QuickCaptureMode: Int] = [:],
        onCapture: @escaping (QuickCaptureMode) -> Void
    ) {
        self.counts = counts
        self.onCapture = onCapture
    }

    public var body: some View {
        let rows = QuickCaptureMode.allCases.chunked(into: 3)

        VStack(alignment: .leading, spacing: AppSpacing.s4) {
            Text("HIGHLIGHT")
                .font(AppText.eyebrow)
                .tracking(AppLetterSpacing.eyebrow)
                .foregroundStyle(AppColors.themeGreenDeep)

            VStack(spacing: 0) {
                ForEach(rows.indices, id: \.self) { rowIndex in
                    if rowIndex > 0 {
                        Divider().background(AppColors.borderDefault)
                    }
                    HStack(spacing: 0) {
                        ForEach(Array(rows[rowIndex].enumerated()), id: \.element) { colIndex, mode in
                            if colIndex > 0 {
                                Divider().background(AppColors.borderDefault)
                            }
                            Pill(
                                mode: mode,
                                count: counts[mode] ?? 0,
                                action: { onCapture(mode) }
                            )
                        }
                    }
                    .fixedSize(horizontal: false, vertical: true)
                }
            }
        }
        .padding(AppSpacing.s5)
        .frame(maxWidth: .infinity, alignment: .leading)
        .background(
            RoundedRectangle(cornerRadius: AppRadius.lg, style: .continuous)
                .fill(AppColors.cardSolid)
        )
        .overlay(
            RoundedRectangle(cornerRadius: AppRadius.lg, style: .continuous)
                .stroke(AppColors.borderDefault, lineWidth: 1)
        )
    }
}

private extension Array {
    func chunked(into size: Int) -> [[Element]] {
        stride(from: 0, to: count, by: size).map {
            Array(self[$0..<Swift.min($0 + size, count)])
        }
    }
}

private struct Pill: View {
    let mode: QuickCaptureMode
    let count: Int
    let action: () -> Void

    var body: some View {
        Button(action: action) {
            VStack(spacing: AppSpacing.s2) {
                Image(systemName: mode.symbol)
                    .font(.system(size: 32, weight: .regular))
                    .foregroundStyle(AppColors.coral)
                    .frame(height: 36)
                Text(mode.label.uppercased())
                    .font(AppText.eyebrow)
                    .tracking(AppLetterSpacing.eyebrow)
                    .foregroundStyle(AppColors.textSecondary)
                    .lineLimit(1)
                Text("\(count)")
                    .font(.system(size: 26, design: .serif))
                    .foregroundStyle(count > 0 ? AppColors.textPrimary : AppColors.textTertiary)
                    .lineLimit(1)
                    .minimumScaleFactor(0.6)
            }
            .padding(.vertical, AppSpacing.s3)
            .frame(maxWidth: .infinity)
        }
        .buttonStyle(.plain)
    }
}
